### PR TITLE
feat(GiniMerchantSDK): Refactoring for images assets

### PR DESCRIPTION
**Pull request by motokotomax**
Proposition to refactor: move all image names to enum, fix naming with common pattern
+ all strings with file names in one place
+ remove optional
+ add fatal error if SDK developer didn't add image resource
+ common prefix to avoid resource collisions 
+ less code in Utils

### CODE REVIEW

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Resources/GiniImages.xcassets/gm.more.imageset/more-vertical (1).pdf
Comment (ValentinaIancu-Gini) [inline MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Resources/GiniImages.xcassets/gm.more.imageset/more-vertical (1).pdf:1]: [clarification]: this file is twice added. is intended like this? also would be nice to have the same name of the actual image and the asset in the catalog?

**Comment by ValentinaIancu-Gini** [comment]
@motokotomax super nice. We already discussed to do same `GiniImage` enum in `GiniBankSDK` for now.
Maybe we should think of doing or at least planning to do the package with shared code asap. I see a lot of duplicated code.

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Resources/GiniImages.xcassets/gm.more.imageset/more-vertical (1).pdf
Comment (razvancapra) [reply MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Resources/GiniImages.xcassets/gm.more.imageset/more-vertical (1).pdf:1]: Yes, one of them is for dark mode, one of them is for light mode. Maybe the naming should be changed a bit.

**Review by razvancapra — APPROVED** [review]
👍  Niceee.

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniImage.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+import UIKit
+
+/**
+ The GiniImage enumeration provides a convenient way to manage image assets within the Gini SDK, supporting customization for both light and dark modes. Each case in the enumeration represents a specific image asset used by the SDK.
+
+ - Note: The raw values for each case correspond to the image asset names in the asset catalog.
+ */
+
+public enum GiniImage: String {
```
Comment (zladzeyka) [inline MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniImage.swift:16]: [Clarification]: GiniImage is quite general and might be used in other places or SDK or even in a shared package that we plan so far, maybe then `GiniMerchantIcon` or `GiniMerchantImage`?

**Review by zladzeyka — COMMENTED** [review]
Looks good to me,
just a tiny clarification 👍🏻

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Resources/GiniImages.xcassets/gm.more.imageset/more-vertical (1).pdf
Comment (motokotomax) [reply MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Resources/GiniImages.xcassets/gm.more.imageset/more-vertical (1).pdf:1]: Hi, thanks for review! Renamed

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniImage.swift (hunk 1 above)
Comment (motokotomax) [reply MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniImage.swift:16]: Thanks for review, done

File: MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniMerchantImage.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+//  GiniImage.swift
+//
+//  Copyright © 2024 Gini GmbH. All rights reserved.
+//
+
+
+import UIKit
+
+/**
+ The GiniImage enumeration provides a convenient way to manage image assets within the Gini SDK, supporting customization for both light and dark modes. Each case in the enumeration represents a specific image asset used by the SDK.
```
Comment (zladzeyka) [inline MerchantSDK/GiniMerchantSDK/Sources/GiniMerchantSDK/Core/GiniMerchantImage.swift:11]:
```suggestion
 The GiniMerchantImage enumeration provides a convenient way to manage image assets within the Gini SDK, supporting customization for both light and dark modes. Each case in the enumeration represents a specific image asset used by the SDK.
```

**Review by zladzeyka — CHANGES_REQUESTED** [review]
@motokotomax Looks good just a tiny suggestion in a code comment

**Review by zladzeyka — APPROVED** [review]
👍🏻 thank you @motokotomax

**Pull request merged**